Tests/PlayerTests/EventsTests.swift

```diff
@@ -154,12 +166,14 @@ final class EventsTests: XCTestCase {
 		player.reset()
 	}
 
-	func testCsssCpbiAndCsseSentAfterSuccessfulLoadAndReset() {
+	func testCsssCpbiAndCsseSentAfterSuccessfulLoadAndReset() async {
 		shouldUseEventProducer = true
 		PlayerWorld.developmentFeatureFlagProvider.shouldSendEventsInDeinit = false
 
 		JsonEncodedResponseURLProtocol.succeed(with: Constants.trackPlaybackInfo)
 
+		await self.asyncSchedulerFactory.executeAll()
```

Comment:
[suggestion] I think in this whole file we don't need the async scheduler since we use a mock and not the actual one (different from `PlayerEventSenderTests`). 🤔 If there were flaky, maybe it is something else we need to look at.

Reply:
Oh, you are right!
I don't know if these specifically were flaky. The `PlayerEventSenderTests` I'm sure. I did it here just in case.
It seems like in `EventsTests` we only need the `optimizedWait()` for the cases where `deinit` is used.